File: skywalking.go
package main

import (
	"context"
	_ "net/http"
	"time"

	// Use this for Elasticsearch 6.x:
	"github.com/olivere/elastic"
)

const DAYS = 7

func main() {
	client, err := elastic.NewClient(elastic.SetURL("http://192.168.1.109:9200"))

	if err != nil {
		// Handle error
		panic(err)
	}

	for _, v := range getIndices() {
		println(v)
	}

	getSum(client, "service_cpm_day-20191113")

	defer client.Stop()
}

func getIndices() []string {
	indices := make([]string, DAYS)
	for i := 0; i < DAYS; i++ {
		t := time.Now().AddDate(0, 0, -i)
		indices[i] = t.Format("service_cpm_day-20060102")
	}
	return indices
}

func getSum(client *elastic.Client, indices string) {
	println("query for data, index=" + indices)

	searchResult, err := client.
		Search().
		Index(indices).
		Query(elastic.NewMatchAllQuery()).
		Size(0).
		Pretty(true).
		Do(context.Background())

	if err != nil {
		panic(err)
	}

	println(searchResult)
}

File: skywalking.py
from elasticsearch import Elasticsearch
import datetime
import time

DAYS = 5


def getIndices():
    indices = []
    begin_date = (datetime.datetime.now() - datetime.timedelta(days=DAYS)).strftime("%Y%m%d")
    begin_date = datetime.datetime.strptime(begin_date, "%Y%m%d")
    end_date = datetime.datetime.strptime(time.strftime('%Y%m%d', time.localtime(time.time())), "%Y%m%d")
    while begin_date < end_date:
        date_str = begin_date.strftime("%Y%m%d")
        indices.append(date_str)
        begin_date += datetime.timedelta(days=1)
    return indices


def getCpm(client, indice):
    response = client.search(
        index=indice,
        body={
            "query": {
                "match_all": {

                }
            },
            "aggs": {
                "x": {"sum": {"field": "total"}}
            }
        })
    return str(int(response['aggregations']['x']['value']))


if __name__ == '__main__':
    client = Elasticsearch(hosts=["http://10.65.3.34:19200/"])
    for i in getIndices():
        print(i + ":" + getCpm(client, "cloud-monitor_service_cpm_hour-" + i))
